src/main/java/org/mockito/internal/creation/bytebuddy/ByteBuddyMockMaker.java

```diff
@@ -37,8 +36,8 @@ public ByteBuddyMockMaker() {
         } catch (ClassCastException cce) {
             throw new MockitoException(join(
                     "ClassCastException occurred while creating the mockito mock :",
-                    "  class to mock : " + describeClass(mockedProxyType),
-                    "  created class : " + describeClass(settings.getTypeToMock()),
+                    "  class to mock : " + describeClass(settings.getTypeToMock()),
+                    "  created class : " + describeClass(mockedProxyType),
```

Comment:
Because they are wrong in the message ;) 
The type to mock is not the class that the mockmaker created ;)